lib/engine/game/g_1849.rb

```diff
@@ -396,13 +409,29 @@ def event_brown_par!
 
       def event_earthquake!
         @log << '-- Event: Messina Earthquake --'
-        # Remove tile from Messina
+        messina = @hexes.find { |h| h.id == 'B14' }
 
-        # Remove from game tokens on Messina
+        city = messina.tile.cities[0]
 
         # If Garibaldi's only token removed, close Garibaldi
+        if (garibaldi = @corporations.find { |c| c.name == 'AFG' })
+          if city.tokened_by?(garibaldi) && garibaldi.placed_tokens.one?
+            @log << '-- AFG loses only token, closing. --'
+            close_corporation(garibaldi)
+          end
+        end
+
+        # Remove from game tokens on Messina
+        @log << '-- Removing tokens from game. --'
+        city.tokens.map(&:destroy!) unless city.tokens.empty?
```

Comment:
empty? not needed as empty list means no destroy anyway

Reply:
`empty?` returns `true` if the array has elements but all are `false` or `nil`, which is the effect I'm looking for here, since the `tokens` array is initialized with `slots`-many `nil`s. Anyway, I changed it to look better.

Reply:
that's not correct, empty only returns true if the element is actually empty, you're looking for tokens.any?

Reply:
city.tokens.each { |token| token&.destroy! }

Reply:
You're right, I had them mixed up. The updated version already includes your suggestion.